spec/translator_spec.lua

```diff
@@ -0,0 +1,341 @@
+local util = require "pallene.util"
+
+local function compile(pallene_code)
+    assert(util.set_file_contents("__test__.pln", pallene_code))
+    local ok, _, _, error_message = util.outputs_of_execute("./pallenec __test__.pln --emit-lua")
+    if not ok then
+        error(error_message)
+    end
+end
+
+local function assert_translation(pallene_code, expected)
+    compile(pallene_code)
+    local contents = util.get_file_contents("__test__.lua")
+    assert.are.same(contents, expected)
+end
+
+local function cleanup()
+    os.remove("__test__.pln")
+    os.remove("__test__.lua")
+end
+
+describe("Pallene to Lua translator", function ()
+    teardown(cleanup)
+
+    it("empty input should result in an empty result", function ()
+        assert_translation("", "")
+    end)
+
+    it("copy the program as is when there are no type annotations", function ()
+        assert_translation([[
+            local i = 10
+            local function print_hello()
+                -- This is a comment.
+                -- This is another line comment.
+                io.write("Hello, world!")
+            end
+        ]],
+        [[
+            local i = 10
+            local function print_hello()
+                -- This is a comment.
+                -- This is another line comment.
+                io.write("Hello, world!")
+            end
+        ]])
+    end)
+
+    it("Remove type annotations from top-level variables", function ()
+        assert_translation([[
+            local xs: integer = 10
+        ]],
+        [[
+            local xs          = 10
+        ]])
+    end)
+
+    it("Keep newlines that appear inside type annotations", function ()
```

Comment:
The newlines in this particular test case are between the colon and the type. Could you add another test case where the newlines are inside the type itself? You could use an array, table or function type to do this.